lib/iris/fileformats/um/_fast_load_structured_fields.py

```diff
@@ -89,9 +90,13 @@ def data(self):
             self._calculate_structure()
         if self._data_cache is None:
             data_arrays = [f._data for f in self.fields]
-            self._data_cache = \
-                ArrayStack.multidim_array_stack(data_arrays,
-                                                self.vector_dims_shape)
+            vector_dims_list = list(self.vector_dims_shape)
```

Comment:
I find this a wee bit confusing.
I wonder if we could usefully employ the '_multidim_daskstack' routine from iris._merge instead ?
It is a recursive rather than iterative algorithm, which is less efficient but maybe clearer, and we would have the advantage that it is done once.  ( Actually I think it lacks unit testing, but that is another issue)
I ***think*** the equivalent would be something like ...
```
data_arrays = np.array([f._data for f in self.fields], dtype=object)
data_arrays = data_arrays.reshape(self.vector_dims_shape)
self._data_cache = _multdim_daskstack(data_arrays)
```